great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -791,13 +798,38 @@ def _build_selectable_from_batch_spec(self, batch_spec) -> Select:
                     .where(split_clause)
                     .limit(batch_spec["sampling_kwargs"]["n"])
                 )
-
+            elif batch_spec["sampling_method"] == "_sample_using_a_list":
+                # <WILL> This check is necessary because of a bug in sqlalchemy that prevents the compilation
+                # of an IN statement using literal_binds. Issue is being tracked by Sqlalchemy https://github.com/sqlalchemy/sqlalchemy/issues/6222
+                # Check if we can remove this entire elif statement in 202105
+                if parse_version(sa.__version__) >= parse_version("1.4.0"):
+                    raise GreatExpectationsError(
+                        f"""
+                        This functionality in Great Expectations version {__version__} is currently incompatible with SqlAlchemy 1.4.0 and higher.
+                        You currently have SqlAlchemy version {sa.__version__}. Please downgrade SqlAlchemy to < 1.4.0 while we work on a proper fix.
+                        """
+                    )
```

Comment:
The only `_sampling_method` that uses a compiled SQL statement that uses the `IN` statement with `literal_binds` is  `_sample_using_a_list()`.  What are your thoughts on adding the check here?

Reply:
this change no longer exists.